File: Mar_PHP/Assignments/BasicPrograms/46.php
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Pascal's Triangle</title>
</head>
<body>
    <h2>Pascal's Triangle</h2>
    <form method="post">
        <label for="rows">Enter the Number Of rows for Pascal Triangle:</label>
        <input type="number" id="rows" name="rows" required>
        <input type="submit" value="Generate Triangle">
    </form>

    <?php
    function getnCr($n, $r) {
        if ($r > $n) {
            return 0;
        }

        if ($r == 0 || $r == $n) {
            return 1;
        }

        return getnCr($n - 1, $r - 1) + getnCr($n - 1, $r);
    }

    if ($_SERVER["REQUEST_METHOD"] == "POST") {
        $rows = $_POST['rows'];

        for ($i = 0; $i < $rows; $i++) {
            for ($space = 0; $space < $rows - $i; $space++) {
                echo " ";
            }
            for ($num = 0; $num <= $i; $num++) {
                echo getnCr($i, $num) . " ";
            }
            echo "<br>";
        }
    }
    ?>
</body>
</html>

File: Jan_Basic_Programming/Assignments/C/46.cpp
#include<iostream>
#include<bits/stdc++.h>
using namespace std;
//Pascal Triangle 

int getnCr(int n, int r){
    if(r>n){
        return 0;
    }

    if(r==0 || r==n){
        return 1;
    }

    return getnCr(n-1, r-1) + getnCr(n-1, r);
}


int main()
{
    int rows;

    cout<<"Enter the Number Of rows for Pascal Triangle : ";
    cin>>rows;

    for(int i=0; i<rows; i++){
        for(int space=0; space<rows-i; space++){
            cout<<" ";
        }
        for(int num=0; num<=i; num++){
            cout<<getnCr(i,num)<<" ";
        }
        cout<<endl;
    }
    
    return 0;
}